--- configure	2012-05-19 18:08:17.000000000 -0700
+++ configure2	2012-05-19 18:08:14.000000000 -0700
@@ -19654,6 +19654,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libtiff" >&5
 $as_echo_n "checking for libtiff... " >&6; }
 
+LIBS="-ltiff -ljpeg $LIBS"
 if test "x${with_libtiff}" = "xyes" -o "x${with_libtiff}" = "x" ; then
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for TIFFScanlineSize64 in -ltiff" >&5
@@ -19662,7 +19663,6 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-ltiff  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
