--- libs/python/src/converter/builtin_converters.cpp	2013-01-09 15:02:42.000000000 -0800
+++ libs/python/src/converter/builtin_converters.cpp	2013-01-09 15:02:44.000000000 -0800
@@ -377,7 +377,8 @@
       static unaryfunc* get_slot(PyObject* obj)
       {
 #if PY_VERSION_HEX >= 0x03000000
-          return (PyUnicode_Check(obj)) ? &py_unicode_as_string_unaryfunc : 0;
+          return (PyUnicode_Check(obj)) ? &py_unicode_as_string_unaryfunc : 
+                  PyBytes_Check(obj) ? &py_object_identity : 0;
 #else
           return (PyString_Check(obj)) ? &obj->ob_type->tp_str : 0;
 
