--- ogr/ogrsf_frmts/openfilegdb/filegdbtable.cpp_	2014-05-31 11:04:40.000000000 -0700
+++ ogr/ogrsf_frmts/openfilegdb/filegdbtable.cpp	2014-05-31 11:05:18.000000000 -0700
@@ -667,7 +667,7 @@
 
     GUInt32 nRemaining = nFieldDescLength - 10;
     nBufferMaxSize = nRemaining;
-    pabyBuffer = (GByte*)VSIMalloc(nBufferMaxSize);
+    pabyBuffer = (GByte*)VSIMalloc(nBufferMaxSize + ZEROES_AFTER_END_OF_BUFFER);
     returnErrorIf(pabyBuffer == NULL );
     returnErrorIf(VSIFReadL(pabyBuffer, nRemaining, 1, fpTable) != 1 );
 
