--- tools/toolutil/ucbuf.c_	2014-05-21 17:56:32.000000000 -0700
+++ tools/toolutil/ucbuf.c	2014-05-21 17:56:36.000000000 -0700
@@ -293,17 +293,6 @@
         }
         outputWritten = (int32_t)(target - pTarget);
 
-
-#if DEBUG
-        {
-            int i;
-            target = pTarget;
-            for(i=0;i<numRead;i++){
-              /*  printf("%c", (char)(*target++));*/
-            }
-        }
-#endif
-
     }else{
         u_charsToUChars(cbuf,target+offset,inputRead);
         outputWritten=((buf->remaining>cbufSize)? cbufSize:inputRead+offset);
