diff --git a/SConstruct b/SConstruct
index cae93d6..53d88e6 100644
--- a/SConstruct
+++ b/SConstruct
@@ -813,6 +813,7 @@ int main()
     return False
 
 def boost_regex_has_icu(context):
+    context.env.Append(LIBS='icui18n')
     ret = context.TryRun("""
 
 #include <boost/regex/icu.hpp>
diff --git a/src/build.py b/src/build.py
index 927cda4..9b92885 100644
--- a/src/build.py
+++ b/src/build.py
@@ -73,6 +73,7 @@ if env['THREADING'] == 'multi':
 if env['RUNTIME_LINK'] == 'static':
     if 'icuuc' in env['ICU_LIB_NAME']:
         lib_env['LIBS'].append('icudata')
+        lib_env['LIBS'].append('icui18n')
 else:
     if env['INTERNAL_LIBAGG']:
           lib_env['LIBS'].insert(0, 'agg')
diff --git a/utils/mapnik-config/build.py b/utils/mapnik-config/build.py
index 8519051..fee9891 100644
--- a/utils/mapnik-config/build.py
+++ b/utils/mapnik-config/build.py
@@ -16,16 +16,16 @@ config_variables = '''#!/bin/sh
 
 ## variables
 
-CONFIG_PREFIX=%(prefix)s
+CONFIG_PREFIX="$( cd "$( dirname $( dirname "${BASH_SOURCE[0]}" ))" && pwd )"
 CONFIG_MAPNIK_LIBNAME=%(mapnik_libname)s
 CONFIG_MAPNIK_INCLUDE=${CONFIG_PREFIX}/include
 CONFIG_MAPNIK_LIB=${CONFIG_PREFIX}/%(libdir_schema)s
 CONFIG_MAPNIK_VERSION='%(version)s'
-CONFIG_MAPNIK_LDFLAGS='%(ldflags)s'
-CONFIG_DEP_LIBS='%(dep_libs)s'
-CONFIG_OTHER_INCLUDES='%(other_includes)s'
-CONFIG_FONTS='%(fonts)s'
-CONFIG_INPUT_PLUGINS='%(input_plugins)s'
+CONFIG_MAPNIK_LDFLAGS=''
+CONFIG_DEP_LIBS=''
+CONFIG_OTHER_INCLUDES="-I${CONFIG_PREFIX}/include -I${CONFIG_PREFIX}/include/freetype2 -DHAVE_JPEG -DBOOST_REGEX_HAS_ICU -DMAPNIK_THREADSAFE"
+CONFIG_FONTS="%(fonts)s"
+CONFIG_INPUT_PLUGINS="%(input_plugins)s"
 CONFIG_GIT_REVISION='%(git_revision)s'
 
 '''
