--- util/domd    2014-06-17 10:32:45.000000000 +0200
+++ util/domd    2014-06-17 10:44:42.000000000 +0200
@@ -12,8 +12,6 @@
 if [ "$MAKEDEPEND" = "" ]; then MAKEDEPEND=makedepend; fi
 
 cp Makefile Makefile.save
-# fake the presence of Kerberos
-touch $TOP/krb5.h
 if expr "$MAKEDEPEND" : '.*gcc$' > /dev/null; then
     args=""
     while [ $# -gt 0 ]; do
@@ -32,7 +30,5 @@
     RC=$?
 fi
 mv Makefile.new Makefile
-# unfake the presence of Kerberos
-rm $TOP/krb5.h
 
 exit $RC
