diff --git a/CMakeLists.txt b/CMakeLists.txt
index a1e733d..b38a38b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,7 +109,7 @@ if(APPLE)
   message(STATUS "Set Architecture to x64 on OS X")
   exec_program(uname ARGS -v  OUTPUT_VARIABLE DARWIN_VERSION)
   string(REGEX MATCH "[0-9]+" DARWIN_VERSION ${DARWIN_VERSION})
-  if(DARWIN_VERSION GREATER 12)
+  if(DARWIN_VERSION GREATER 12 AND NOT OSXLIBSTD)
     message(STATUS "Activating -std=c++11 flag for >= OS X 10.9")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
   endif()
