diff --git a/osx/out/packages/protobuf-2.5.0-arm64/src/google/protobuf/stubs/platform_macros.h b/osx/out/packages/protobuf-2.5.0-arm64/src/google/protobuf/stubs/platform_macros.h
index b1df60e..afe18a0 100644
--- a/osx/out/packages/protobuf-2.5.0-arm64/src/google/protobuf/stubs/platform_macros.h
+++ b/osx/out/packages/protobuf-2.5.0-arm64/src/google/protobuf/stubs/platform_macros.h
@@ -57,6 +57,9 @@
 #elif defined(__ppc__)
 #define GOOGLE_PROTOBUF_ARCH_PPC 1
 #define GOOGLE_PROTOBUF_ARCH_32_BIT 1
+#elif defined(__aarch64__)
+#define GOOGLE_PROTOBUF_ARCH_ARM 1
+#define GOOGLE_PROTOBUF_ARCH_64_BIT 1
 #else
 #error Host architecture was not detected as supported by protobuf
 #endif
