Index: tests/run_tests.py
===================================================================
--- tests/run_tests.py	(revision 2488)
+++ tests/run_tests.py	(working copy)
@@ -46,7 +46,8 @@
 
     if not quiet:
         print "- mapnik path: %s" % mapnik.__file__
-        print "- _mapnik.so path: %s" % mapnik._mapnik.__file__
+        if hasattr(mapnik,'_mapnik'):
+           print "- _mapnik.so path: %s" % mapnik._mapnik.__file__
         print "- Input plugins path: %s" % mapnik.inputpluginspath
         print "- Font path: %s" % mapnik.fontscollectionpath
         print
